include/bbp/sonata/output.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <algorithm>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace {
+template <class InputIt1, class InputIt2, class OutputIt, class BinaryOperation, class BinaryPredicate>
+OutputIt transform_if(InputIt1 first1, InputIt1 last1, InputIt2 first2, OutputIt d_first, BinaryOperation binary_op, BinaryPredicate binary_pred) {
+    while (first1 != last1) {
+        if (binary_pred(*first1, *first2)) {
+            *d_first++ = binary_op(*first1, *first2);
+        }
+        first1++, first2++;
```

Comment:
pre-increment to avoid the copy?  Also, please put them on separate lines. (I may need to override the `operator,`  some day /sarcasm)

Reply:
Done